### DISCUSSION TITLE
Night time lapse with auto shutter speed and ISO<800 is possible?

### DISCUSSION THREAD

**Discussion by Evgenii063** [gopro/labs#971] [Q&A] [answered]
Hello everyone, friends!  Tell me, I encountered the same problem.  How to shoot night timelapse on gopro 11 black - from day to night?  Namely, I want the time-lapse night mode to be able to raise the ISO above 800 with automatic shutter speed.  Now in the camera you can only set the ISO value above 800, only with shutter speed = 5,10,15,20,30 seconds.  Roughly speaking, how to make the following settings: shutter speed = auto, iso min =100, iso max <800?

**Comment by dnewman-gpsw** [comment, marked as answer]
Night-lapse Auto shutter will transition from day to night very well, going for daylight 1/1500 @ 100 ISO to 10s at 800 ISO automatically.  The only issue if you are in a dark sky location and want see the Milky Way, which needs 20-30s exposures are 3200 or 1600 ISO.  For this you will need to script to camera to switch from auto to fixed exposure at a particular time or at a particular light level.  This will gave separate timelaspes.

Untested example script using the 800 level as a switch.
![image](https://github.com/gopro/labs/assets/10856084/4db1d5b2-efe4-4cc5-b5c0-0c837b2e7b87)
  
![Auto Nightlapse-6_27_2024, 11_39_59](https://github.com/gopro/labs/assets/10856084/90d5301e-84f0-41f3-b1bb-d4f967ae6e67)

**Reply by Evgenii063** [reply]
thank you friend !  can I also ask, if I scan your QR code, then how can I disable it?  can you give me the qr code?  I'm not good at writing such scripts, but you're great!

**Reply by dnewman-gpsw** [reply]
The QR code in the above.  The script will stop when you just press stop recording.

**Reply by Evgenii063** [reply]
friend, thanks for your help and your answer.

**Reply by Evgenii063** [reply]
this code does not work on hero 11. can anyone give a ready-made solution?

**Reply by dnewman-gpsw** [reply]
It works on HERO11, so it is a ready made solution.  What are you experiencing?  Are you running the latest Labs firmware?

**Reply by Evgenii063** [reply]
yes, the latest version of the laboratory, I want to shoot a timelapse from day to morning so that the Milky Way and stars can be clearly seen.  and at the same time so that it is clearly visible during the day (so that the shutter speed does not spoil daytime photographs from the sun)

**Reply by Evgenii063** [reply]
When using this QR code, the camera takes one picture, then there is a pause of 60 seconds and the shooting stops.  Nothing further happens.

**Reply by dnewman-gpsw** [reply]
Picture?  A photo?  The QR Code is creating a Video Night Lapse, it will work only if you have a Night Lapse preset set to video (the default.)  Did you only want photos?

**Reply by Evgenii063** [reply]
It is in night mode with the video value that I use this qr code, but it does not work.  (I scan the QR code, 60 seconds count down and after that nothing happens) the camera is in standby mode.

**Reply by dnewman-gpsw** [reply]
As Labs is technical you will have learn the steps to see what you are doing or have setup wrong. As this works on multiple cameras is my testing.

Start with 
![image](https://github.com/gopro/labs/assets/10856084/7e9e6db1-8ea6-45e2-a521-cca612ee0dfc)

What does your camera do?

**Comment by Evgenii063** [comment]
It just goes into night mode.  according to this qr code

**Reply by dnewman-gpsw** [reply]
Does you camera enter Night Lapse Video mode?  

What does this next QR do? It is just the first part of the script:

![image](https://github.com/gopro/labs/assets/10856084/e0d287b3-b089-48ec-8c26-b802529723c7)

**Reply by Evgenii063** [reply]
The camera switches to night time-lapse video mode.  when scanning the qr code, the camera starts recording time-lapse video with the settings: shutter speed = auto, ISO MIN = 100, ISO MAX = 800, frame interval 20 seconds.

**Reply by dnewman-gpsw** [reply]
So it is working so far. You should experiment to see where it is stopping.

**Reply by Evgenii063** [reply]
stops just at the following QR code configurations, after a 60 second countdown.

**Reply by dnewman-gpsw** [reply]
There is no countdown in the script, there is a sleep 60, yet in the camera is recording that will be silent. Please film the camera to show the error. Also start in daylight for the test.

**Reply by Evgenii063** [reply]
friend, this is what happens to the camera after scanning your QR code.  I made a video - take a look. https://youtu.be/nPOcLqaQ88g?feature=shared

**Reply by dnewman-gpsw** [reply]
When you wrote "changed your code, now it works"  All you did was change 20 interval to 5 seconds.  So basically the same script is working. But based on the video it appears to behaving almost correctly.  

I added more exposure math. 
![Auto Nightlapse-6_29_2024, 17_22_18](https://github.com/gopro/labs/assets/10856084/b798430e-4b6e-4b7b-9914-a7d3788d28d8)

**Reply by Evgenii063** [reply]
friend, can you also send me the command tree here?  I want to see what you changed and how you changed it to take into account the mistakes and become a stronger user)))

**Reply by dnewman-gpsw** [reply]
It is the metadata of the PNG attached above, download with "Save Link As..." then load the image into Blockly using Import Image.

**Reply by Evgenii063** [reply]
I see, thanks bro!

**Reply by Evgenii063** [reply]
friend, I want to upset you, but this QR code doesn’t work either.  when scanning a qr code, it turns on recording with an interval of 5 seconds with the parameters: shutter speed = auto < iso min = 100, iso max = 800, that is, in a dark room the parameter with a shutter speed of 20 seconds does not turn on (((

**Reply by Evgenii063** [reply]
This is probably not feasible after all.  too difficult for the camera.   Let's enjoy the standard functions of the camera.  Thanks friend for your help!  I appreciate it !  :)

**Reply by dnewman-gpsw** [reply]
Working fine here. I used a lens cap to simulate required darkness in my testing.  When the Gain is greater 25, the camera switches to the longer exposures. The 5 seconds mode the maximum gain is 40 (ISO 800/100*5seconds.)  You can watch the current gain and ISO on the LCD to see it working. You now have all the tools you need to tune the to your environment.

Here is a timelapse of the timelapse, to show it is working.  Effectively the video is speed up 15x.

https://github.com/gopro/labs/assets/10856084/d900f97c-157f-4959-bb40-cc697fcdca7a

**Reply by Evgenii063** [reply]
what qr code was used in this video?

**Reply by dnewman-gpsw** [reply]
Same one I posted 16 hours ago (I scanned it from the posting.)

**Comment by Evgenii063** [comment]
One more question, is it possible to use the night photo mode (single photo) to take photos after a period of time according to a script?

**Reply by dnewman-gpsw** [reply]
Anything time based is much easier than exposure based.  Yes, this would be straight forward.
